### ISSUE TITLE
LGTM.com - false positive

### ISSUE THREAD

**Issue by ljharb** [Semmle/ql#2621]
**Description of the false positive**

Strict mode can not use `arguments.callee`, this is true, but i'm using it inside a try/catch explicitly to *trigger* the exception.

https://lgtm.com/projects/g/ljharb/es-abstract/snapshot/a0800679d508ef15d31fe84283fc6a522277ca23/files/GetIntrinsic.js?sort=name&dir=ASC&mode=heatmap#xbd19a52cddeabb11:1

It's also marked as "has no effect", but it throws an exception in strict mode code, which is quite a strong effect.

https://lgtm.com/projects/g/ljharb/es-abstract/snapshot/a0800679d508ef15d31fe84283fc6a522277ca23/files/GetIntrinsic.js?sort=name&dir=ASC&mode=heatmap#xbd19a52cddeabb11:1

In other words, I think that *both* of these should never show up for `arguments.callee`. In sloppy mode, this is a noop expression; in strict mode, it has an effect (and ideally it should be permitted as a noop expression inside a try/catch)

**Comment by asgerf** [comment]
Hi ljharb,

Thanks for the report. https://github.com/Semmle/ql/pull/2627 and https://github.com/Semmle/ql/pull/2624 should fix the two spurious alerts. The alerts should go away next time LGTM.com updates.

**Comment by ljharb** [comment]
Thank you!
